presto-main/src/main/java/io/prestosql/dispatcher/FailedDispatchQuery.java

```diff
@@ -226,6 +226,7 @@ private static QueryInfo immediateFailureQueryInfo(
                 ImmutableList.of(),
                 ImmutableSet.of(),
```

Comment:
There are missing information about accessed:
 - procedures (must have)
 - functions (must have)
 - DDL queries 
 - set session 
 - SHOW queries
 - query checks

All these surely should go as separate pull requests.

Reply:
DDL queries can be inferred from the query type + user + target table. Session properties can be inferred from the session properties in the query + user. Functions and procedures can be added, but I'll do it in a separate PR.